metamodel.zenta

```diff
@@ -2335,180 +2365,1111 @@ A role is assigned to human(s).
+* simple constants are public static final fields in SCREAMING_SNAKE_CASE
+* in the production code there are no constants with complex structures
+</documentation>
+            </element>
+            <element xsi:type="zenta:BasicObject" name="data structure unit" id="b585a667-63bb-4bb8-8a20-2e574b874ed0" ancestor="behaviour">
+              <documentation>data structures are defined in DTOs. And Entity is a special kind of DTO.
+data structure units do not have behaviours (no methods)
+
+</documentation>
+            </element>
+            <element xsi:type="zenta:BasicObject" name="framework specific" id="6cd0ac90-c94f-4779-afe5-c704590846d8" ancestor="behaviour">
+              <documentation>Configuration is provided by means specific to the framework used.
+For example in Java Spring projects the SpringConfig Class contains the configuration variables as fields.
+</documentation>
+            </element>
+            <element xsi:type="zenta:BasicObject" name="generated" id="35eddd23-6029-42b8-9bb9-b763e46976e3" ancestor="behaviour">
+              <documentation>* Where all the information necessary to write a unit exists in the model, the unit will be generated in the long term
+* If such a unit is missing, it is the responsibility of the coder to create it.
+
+</documentation>
+            </element>
+            <element xsi:type="zenta:BasicObject" name="helper" id="a83b3a77-8854-44b0-977e-037c7864d052" ancestor="behaviour">
+              <documentation>* Helpers are put to the same package as the code they support.
+* If the supported units are in different packages, then the helper is put to the lowest common superpackage.
+* A helper unit is created by the coder if code complexity constraints or duplicate avoidance makes it necessary.
+
+</documentation>
+            </element>
+            <element xsi:type="zenta:BasicObject" name="interface class" id="2a9d2132-6761-4332-90ef-5867fc9d6096" ancestor="behaviour">
+              <documentation>* Units without naming convention (Delegate, Configuration provider, and units which part of the boilerplate needed with the frameworks used) should have an @InterfaceClass annotation, denoting the lack of naming convention applicable.</documentation>
+            </element>
+            <element xsi:type="zenta:BasicObject" name="modeled placement" id="6a6a5f4e-be42-4cad-bf72-be0116b8817e" ancestor="behaviour">
+              <documentation>First-class production code units are placed in the production code hierarchy under the package designated by the design
+</documentation>
+            </element>
+            <element xsi:type="zenta:BasicObject" name="No reference to production code constants" id="5397f74e-8346-4361-a715-1118727df296" ancestor="behaviour">
+              <documentation>Test units must not reference constants in production code in any way. If a constant is needed, it should be provided by test data.</documentation>
+            </element>
+          
```

Comment:
What about languages where no display name feature exists? Wouldn't it be simpler to use a descriptive test method name?

Reply:
We can add an annotation or other language construct supported by the language. In worst case it can be a specially formatted comment. With function names we don't have the flexibility for punctuations and such.
For javascript consider this QUnit example, which is okay for me:

```
test("This is the name for this test", function () { 
    ok(true);	
});
```